File: backend/app/schemas/produccion.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class SesionCreate(BaseModel):
    fecha: date
    turno: str = Field(..., pattern="^(manana|tarde)$")


class SesionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    finca_id: str
    fecha: date
    turno: str
    operario_id: str
    created_at: datetime
    updated_at: datetime


class RegistroProduccionCreate(BaseModel):
    animal_id: str
    volumen_litros: float = Field(..., gt=0, description="Litros con un decimal")


class RegistroProduccionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    finca_id: str
    sesion_id: str
    animal_id: str
    fecha: date
    turno: str
    volumen_litros: float
    operario_id: str
    created_at: datetime


class ProduccionDiariaRead(BaseModel):
    fecha: date
    total_litros: float

File: frontend/src/types/produccion.ts
export type Turno = "manana" | "tarde";

export interface Sesion {
  id: string;
  finca_id: string;
  fecha: string;
  turno: Turno;
  operario_id: string;
  created_at: string;
  updated_at: string;
}

export interface RegistroProduccion {
  id: string;
  finca_id: string;
  sesion_id: string;
  animal_id: string;
  fecha: string;
  turno: Turno;
  volumen_litros: number;
  operario_id: string;
  created_at: string;
}

export interface RegistroProduccionCreate {
  animal_id: string;
  volumen_litros: number;
}

export interface ProduccionDiaria {
  fecha: string;
  total_litros: number;
}

export interface Alerta {
  id: string;
  finca_id: string;
  animal_id: string;
  tipo_alerta: "caida_produccion" | "sin_registro_24h";
  nivel: "critico" | "advertencia";
  mensaje: string;
  revisada: boolean;
  fecha_revision: string | null;
  datos_extra: Record<string, unknown> | null;
  created_at: string;
  animal_nombre: string | null;
  animal_numero_arete: string | null;
}
